src/rdkafka_msgset_reader.c

```diff
@@ -902,6 +1022,23 @@ rd_kafka_msgset_reader_v2 (rd_kafka_msgset_reader_t *msetr) {
                                               &save_slice, payload_size))
                         rd_kafka_buf_check_len(rkbuf, payload_size);
 
+                if (msetr->msetr_aborted_txn_offsets != NULL &&
+                    msetr->msetr_v2_hdr->Attributes & RD_KAFKA_MSGSET_V2_ATTR_TRANSACTIONAL &&
+                    !(msetr->msetr_v2_hdr->Attributes & RD_KAFKA_MSGSET_V2_ATTR_CONTROL)) {
+
+                        rd_kafka_aborted_txn_start_offsets_t node, *node_ptr;
+                        node.pid = msetr->msetr_v2_hdr->PID;
+                        node_ptr = RD_AVL_FIND(msetr->msetr_aborted_txn_offsets, &node);
+
+                        if (node_ptr &&
+                            node_ptr->offsets_idx < rd_list_cnt(&node_ptr->offsets) &&
+                            msetr->msetr_v2_hdr->BaseOffset >= *(int64_t *)rd_list_elem(&node_ptr->offsets, node_ptr->offsets_idx)) {
```

Comment:
observe 80 cols

Reply:
technically, that applies in many more places as well and seems really awkward to strictly enforce.
i'll make this line shorter and wait for further complaints before making more changes.